src/components/ImageUploadModal.tsx

```diff
@@ -109,6 +110,10 @@ function ImageUploadModal({ visible, onClose }) {
 
     const previewWindow = window.open(thumbUrl, '_blank');
 
+    if (previewWindow === null) {
+      throw new Error('Could not open preview window.');
```

Comment:
Probably best to just log to the console and return (or maybe show an error notification). This would occur if `window.open` gets blocked by the client, throwing an error could result in undesired results